pkg/ui/ccl/src/views/reports/containers/stores/encryption.tsx

```diff
@@ -14,16 +28,77 @@ export default class EncryptionStatus extends React.Component<EncryptionStatusPr
     );
   }
 
-  render(): React.ReactElement<any> {
+  renderKey(isStoreKey: boolean, key: protos.cockroach.ccl.storageccl.engineccl.enginepbccl.KeyInfo$Properties) {
+    // Get the enum name from its value (eg: "AES128_CTR" for 1).
+    const encryptionType = protos.cockroach.ccl.storageccl.engineccl.enginepbccl.EncryptionType[key.encryption_type];
+    const createdAt = moment.unix(key.creation_time.toNumber()).utc().format(dateFormat);
+
+    if (isStoreKey) {
+      return [
+        this.renderHeaderRow("Active Store Key: user specified"),
+        this.renderSimpleRow("Algorithm", encryptionType),
+        this.renderSimpleRow("Key ID", key.key_id),
+        this.renderSimpleRow("Created", createdAt),
+        this.renderSimpleRow("Source", key.source),
+      ];
+    } else {
+      return [
+        this.renderHeaderRow("Active Data Key: automatically generated"),
+        this.renderSimpleRow("Algorithm", encryptionType),
+        this.renderSimpleRow("Key ID", key.key_id),
+        this.renderSimpleRow("Created", createdAt),
+        this.renderSimpleRow("Parent Key ID", key.parent_key_id),
+      ];
+    }
+  }
+
+  renderFileStats(stats: protos.cockroach.server.serverpb.StoreDetails$Properties) {
+    if (stats.total_files.eq(0) && stats.total_bytes.eq(0)) {
+      return null;
+    }
+
+    let percentFiles = 100;
+    if (stats.active_key_files !== stats.total_files) {
+      percentFiles = Long.fromInt(100).mul(stats.active_key_files).toNumber() / stats.total_files.toNumber();
```

Comment:
why do some arithmetic in one data type and the rest of the arithmetic in a different data type?

Reply:
You mean do the multiplication using Long and the division using floats? I need longs for multiplications for accuracy, but I need floats for division otherwise I get `33.00%` instead of `33.33%`